doc/arkode/guide/source/Usage/LSRKStep/User_supplied.rst

```diff
@@ -0,0 +1,45 @@
+.. ----------------------------------------------------------------
+   Programmer(s): Daniel R. Reynolds @ SMU
+                  David J. Gardner @ LLNL
+   ----------------------------------------------------------------
+   SUNDIALS Copyright Start
+   Copyright (c) 2002-2024, Lawrence Livermore National Security
+   and Southern Methodist University.
+   All rights reserved.
+
+   See the top-level LICENSE and NOTICE files for details.
+
+   SPDX-License-Identifier: BSD-3-Clause
+   SUNDIALS Copyright End
+   ----------------------------------------------------------------
+
+.. _LSRKSTEP.Usage.UserSupplied:
+
+User-supplied functions
+=============================
+
+Currently, there is only one user-supplied function for LSRKStep (in addition to the :c:type:`ARKRhsFn` arguments that define the IVP).
+
+
+
+
+.. _LSRKStep.Usage.DomEig:
+
+The dominant eigenvalue estimation
+----------------------------------
+
+When running LSRKStep with either the RKC or RKL methods, the user must supply a dominant eigenvalue estimation function of type :c:type:`ARKDomEigFn`:
+
+.. c:type:: int (*ARKDomEigFn)(sunrealtype* t, N_Vector y, sunrealtype* lambdaR, sunrealtype* lambdaI, void* user_data)
```

Comment:
Since we plan on adding complex support later with `suncomplextype`, I wonder if we should add `suncomplextype` to `sundials_types.h` now so it can be used in this function (instead of having a `lambdaR` and `lambdaI`). What do you think @gardner48 @drreynolds ?

Reply:
we need to use this even in the real valued problems, `suncomplextype` might not be appropriate

Reply:
Having a single complex output makes sense to me (I made a similar comment in my pass over the docs)

Reply:
I was thinking that this would always be available. Adding the type should be quick, it only requires adding a few lines to the `sundials_types.h`.